crates/sui-framework/packages/dwallet-system/sources/dwallet.move

```diff
@@ -310,4 +310,27 @@ module dwallet_system::dwallet {
         };
         transfer::transfer(sign_output, session.sender);
     }
+
+    /// Event to start a `MockMPCSession`, caught by the Validators.
+    struct CreatedProofMPCSessionEvent has copy, drop {
+        session_id: ID,
+        sender: address,
+    }
+
+    struct ProofSessionData has key {
+        id: UID,
+    }
+
+    /// Function to launch proof MPC flow.
+    public fun launch_proof_mpc_flow(ctx: &mut TxContext) {
+        let session_data = ProofSessionData {
+            id: object::new(ctx),
+        };
+        let created_proof_mpc_session = CreatedProofMPCSessionEvent {
+            session_id: object::id(&session_data),
```

Comment:
Does this follow specification? Please document why you did this, with reference to the paper.

Reply:
Also annotate